provider/cluster/monitor.go

```diff
@@ -176,10 +177,10 @@ func (m *deploymentMonitor) doCheck() (bool, error) {
 	return badsvc == 0, nil
 }
 
-func (m *deploymentMonitor) runCloseLease() <-chan runner.Result {
+func (m *deploymentMonitor) runCloseLease(ctx context.Context) <-chan runner.Result {
 	return runner.Do(func() runner.Result {
 		// TODO: retry, timeout
-		err := m.session.Client().Tx().Broadcast(context.Background(), &mtypes.MsgCloseBid{
+		err := m.session.Client().Tx().Broadcast(ctx, &mtypes.MsgCloseBid{
```

Comment:
we might have to do a reaper task that closes leases in addition to the one we've talked about that cleans up namespaces.